### PR TITLE
fix(readonly-confirmation): fix read-only prefix error

### PR DESCRIPTION
**Pull request by Tongky-HGU**
#### Summary

- fix CSS error in `:read-only` example

#### Motivation

- `:read-only` CSS is not working properly in Chorme.
- `-moz-` prefix won't work with Chome browser.
- Chome ignores whole CSS block with invalid prefix.

*As-Is*
image
*To-Be*
image

#### Supporting details

> A selector always goes together with a declaration block.
> When a user agent cannot parse the selector (i.e., it is not valid CSS 2.1),
> it must ignore the selector and the following declaration block (if any) as well.
> &mdash; [4.1.7 Rule sets, declaration blocks, and selectors](<https://www.w3.org/TR/CSS21/syndata.html#rule-sets:~:text=A%20selector%20always%20goes%20together%20with%20a%20declaration%20block.%20When%20a%20user%20agent%20cannot%20parse%20the%20selector%20(i.e.%2C%20it%20is%20not%20valid%20CSS%C2%A02.1)%2C%20it%20must%20ignore%20the%20selector%20and%20the%20following%20declaration%20block%20(if%20any)%20as%20well.>)

- [autoprefixer](https://autoprefixer.github.io/) result
```css
/* input */
input:read-only,
textarea:read-only {
  border: 0;
  box-shadow: none;
  background-color: white;
}
```

```css
/* output */
input:-moz-read-only, textarea:-moz-read-only {
  border: 0;
  box-shadow: none;
  background-color: white;
}
input:read-only,
textarea:read-only {
  border: 0;
  -webkit-box-shadow: none;
          box-shadow: none;
  background-color: white;
}
```

### CODE REVIEW

**Comment by teoli2003** [comment]
>     * `:read-only` CSS is not working properly in Chrome.

Can you elaborate on this? According to the browser compatibility data `:read-only` is supported since Chrome 1.

Also, Firefox supports `:read-only` without prefix for 2 years. 

I would prefer to remove all prefixes in this case.

File: html/forms/pseudo-classes/readonly-confirmation.html [hunk 1]
```diff
@@ -61,18 +61,31 @@
...
         background-color: white;
       }
 
-      textarea:-moz-read-write,
-      textarea:read-write {
+      input:read-only,
+      textarea:read-only {
+        border: 0;
+        -webkit-box-shadow: none;
+                box-shadow: none;
```
Comment (teoli2003) [inline html/forms/pseudo-classes/readonly-confirmation.html:75]: Am I wrong in saying no browser support unprefixed `box-shadow`?

File: html/forms/pseudo-classes/readonly-confirmation.html [hunk 2]
```diff
@@ -61,18 +61,31 @@
...
+      }
+
+      textarea:-moz-read-write {
         box-shadow: inset 1px 1px 3px #ccc;
         border-radius: 5px;
       }
+
+      textarea:read-write {
+        -webkit-box-shadow: inset 1px 1px 3px #ccc;
+                box-shadow: inset 1px 1px 3px #ccc;
```
Comment (teoli2003) [inline html/forms/pseudo-classes/readonly-confirmation.html:86]: Same here

**Review by teoli2003 — CHANGES_REQUESTED** [review]

**Comment by teoli2003** [comment]
@teoli2003, @wbamberg What is the policy with added an unprefixed CSS property when no browser supports it unprefixed (and all, including Gecko, with the `-webkit` prefix?

**Comment by Tongky-HGU** [comment]
I think my PR is misleading. `:-moz-read-only` blocks whole css block include `:read-only`, so example is not working properly in Chorme.

I believe that all the above questions are discussed here. -> https://github.com/mdn/content/pull/17200

**Review by teoli2003 — APPROVED** [review]

**Comment by vlakoff** [comment]
I've just added a comment on the sibling ticket: https://github.com/mdn/content/pull/17200#issuecomment-1888547101.

**Pull request merged**